assert/assertions_test.go

```diff
@@ -100,47 +100,42 @@ func (a *AssertionTesterConformingObject) TestMethod() {
 type AssertionTesterNonConformingObject struct {
 }
 
-func TestObjectsAreEqual(t *testing.T) {
+// TestCase holds the expected/actual values to be passed to most checks and their expected result (true/false)
+type TestCase struct {
+	expected interface{}
+	actual   interface{}
+	result   bool
+}
```

Comment:
I'm not sure this provides much value as it's incredibly abstract. It basically says 'a test case can expect anything, compared with anything, and the result can be success or fail' :)

In my opinion it's better to just do them on an individual basis (i.e. per-test) and actually have them be specific to the function we're testing. Thoughts?

Reply:
[This](https://github.com/golang/go/wiki/TableDrivenTests#example-of-a-table-driven-test) tends to be the _de facto_ template for table-driven tests in Go. In my opinion we should try and stick to it :)

Reply:
It's modeled after the signature of most asserts - they take an expected, an actual and return a (bool) result.

I'm fine with defining them on  per assert-test basis but it will still look mostly the same (except that you don't have to provide a name for the struct anymore - it can be entirely inline)

Also, following the "de facto template" looks good, but that change is mosty about wrapping the test in a t.Run() (which would probably need a descriptive string per test to properly name the func)